teuthology/orchestra/run.py

```diff
@@ -221,6 +238,63 @@ def __repr__(self):
             name=self.hostname,
             )
 
+def find_unittest_error(xmlfile_path: str, client):
+    """
+    Parse the xml file.
+    If xmlfile_path is dir, parse all xml files.
+    """
+    if not xmlfile_path:
+        return "No XML file path was passed to process!"
+    if xmlfile_path[-1] == "/": # directory
+        (_, stdout, _) = client.exec_command(f'ls -d {xmlfile_path}*.xml', timeout=200)
+        xml_files = stdout.read()
+        xml_files = xml_files.split('\n')
+        for file in xml_files:
+            error = parse_xml(file, client)
+            if error:
+                return error
+    elif os.path.splitext(xmlfile_path)[1] == ".xml": # xml file
+        error = parse_xml(xmlfile_path, client)
+        return error
+
```

Comment:
nit: for better readability instead of returning error in every if, elif condition, maybe we can just return ``error`` at the very end. Declare error = None at the top.